libraries/botbuilder-dialogs/botbuilder/dialogs/skills/skill_dialog.py

```diff
@@ -226,8 +228,86 @@ async def _send_to_skill(
+                    pass
                 else:
                     # Send the response back to the channel.
                     await context.send_activity(from_skill_activity)
 
         return eoc_activity
+
+    async def _intercept_oauth_cards(
+        self, context: TurnContext, activity: Activity, connection_name: str
+    ):
+        """
+        Tells is if we should intercept the OAuthCard message.
+        """
+        if not connection_name or not isinstance(
+            context.adapter, ExtendedUserTokenProvider
+        ):
+            return False
+
+        oauth_card_attachment = next(
+            attachment
+            for attachment in activity.attachments
+            if attachment.content_type == ContentTypes.oauth_card
+        )
+        if oauth_card_attachment:
+            oauth_card = oauth_card_attachment.content
+            if (
+                oauth_card
+                and oauth_card.token_exchange_resource
+                and oauth_card.token_exchange_resource.uri
+            ):
+                try:
+                    result = await context.adapter.exchange_token(
+                        turn_context=context,
+                        connection_name=connection_name,
+                        user_id=context.activity.from_property.id,
+                        exchange_request=TokenExchangeRequest(
+                            uri=oauth_card.token_exchange_resource.uri
+                        ),
+                    )
+
+                    if result and result.token:
+                        return await self._send_token_exchange_invoke_to_skill(
+                            activity,
+                            oauth_card.token_exchange_resource.id,
+                            oauth_card.connection_name,
+                            result.token,
+                        )
+                except:
+                    return False
+
+        return False
+
+    async def _send_token_exchange_invoke_to_skill(
+        self,
+        incoming_activity: Activity,
+        request_id: str,
+        connection_name: str,
+        token: str,
+    ):
+        activity = incoming_activity.create_reply()
+        activity.type = ActivityTypes.invoke
+        activity.name = SignInConstants.token_exchange_operation_name
+        activity.value = TokenExchangeInvokeRequest(
+            id=request_id, token=token, connection_name=connection_name,
+        )
+
+        # route the activity to the skill
+        skill_info = self.dialog_options.skill
+        response = await self.dialog_options.skill_client.post_activity(
+            self.dialog_options.bot_id,
+            skill_info.app_id,
+            skill_info.skill_endpoint,
+            self.dialog_options.skill_host_endpoint,
+            incoming_activity.conversation.id,
+            activity,
+        )
+
+        # Check response status: true if success, false if failure
+        return response.status / 100 == 2
```

Comment:
First time I see this, interesting. Is there no built in or popular function in python for this? i.e. dotnet has HttpStatusCodes and IsSuccessStatusCode() in a package that gets added everywhere anyways. If it requires adding a new package for this check, then it is not worth, but just checking.